api/rest/client/methods.go

```diff
@@ -86,6 +86,42 @@ func (c *defaultClient) Unpin(ci cid.Cid) error {
 	return c.do("DELETE", fmt.Sprintf("/pins/%s", ci.String()), nil, nil, nil)
 }
 
+// PinPath resolves given path into a cid and performs the pin operation.
+func (c *defaultClient) PinPath(pinPath string, replicationFactorMin, replicationFactorMax int, name string) (cid.Cid, error) {
+	var ci api.CidSerial
+
+	escName := url.QueryEscape(name)
+	err := c.do(
+		"POST",
+		fmt.Sprintf(
+			"/pins/%s?replication-min=%d&replication-max=%d&name=%s",
+			api.TrimSlacesAndSpaces(pinPath),
```

Comment:
do not modify the url path. Things should error (404) by default if the path is too bad to match the API endpoint. A path is not a free-form string where everything should be allowed (like `name`).